x-pack/plugins/maps/public/util.ts

```diff
@@ -71,5 +77,14 @@ export function makePublicExecutionContext(
   id: string,
   description?: string
 ): KibanaExecutionContext {
-  return makeExecutionContext(id, window.location.pathname, description);
+  const topLevelContext = getExecutionContext().get();
+  const context = makeExecutionContext(id, window.location.pathname, description);
+
+  // Distinguish between running in maps app vs. embedded
+  return topLevelContext.name !== context.name
```

Comment:
This check is not working as intended. topLevelContext.name is undefined so child is set every time, even when in the maps application

<img width="400" alt="Screen Shot 2022-02-23 at 11 20 36 AM" src="https://user-images.githubusercontent.com/373691/155382750-b13ecf28-65eb-4018-9de6-57eecf49eaee.png">

Reply:
Thank you - I will try to implement a jest test for this for future stabilization!

Reply:
Fixed!

Reply:
Does this look like another side issue of making all fields optional? https://github.com/elastic/kibana/pull/124996/files#r808273840: